### ISSUE TITLE
Problem fetching subscription SKU's via getSubscription method in useIAP hook

### ISSUE THREAD

**Issue by cbothra** [dooboolab/react-native-iap#1601]
Hi, I am stuck at an issue and it still persists even after doing a lot of research to fix it. Problem is I am not able to fetch the subscriptions from app-store connect. However when I use Storekit configuration, the library works fine as expected in both emulator and real-device. 

I also cross checked that my tax & agreements are signed, agreed and are active. Subscription sku's state is ready-to-submit. I am not sure what key thing I am missing here. My Xcode version is 13.2.1

Any help would be highly appreciated.

### Version of react-native-iap
7.5.3

### Version of react-native
0.63.3

### Platforms you faced the error (IOS or Android or both?)
IOS

### Expected behavior
After successfully getting the useIAP hook's connected status as true, I shall be able to get the subscriptions products in the array.

### Actual behavior
Getting an empty array of subscriptions.

### Tested environment (Emulator? Real Device?)
Real Device.

**Comment by cbothra** [comment]
Is there anything like that in order to test IAP in the sandbox mode, the Storekit.configuration is the only way. We cannot directly interact with App Store Connect?

**Comment by siddharthpvi** [comment]
Hi Can you help me for getting product array on iOS Side?

**Comment by cbothra** [comment]
Sure, if you can tell exact issue which you are facing. Happy to help.

**Comment by siddharthpvi** [comment]
Yes sure, I create 1 Subscription on the Apple store.  Now try to getting the product array using this code

 RNIap.initConnection()
      .catch(() => {
        console.log("error connecting to store...");
      })
      .then(() => {
        RNIap.getSubscriptions(items)
          .catch(() => {
            console.log("error finding items");
          })
          .then((res) => {
            console.log("Resss", JSON.stringify(res));

            setProducts(res);
          });

And it give me error 
and in Android side this is working fine

**Comment by cbothra** [comment]
What is the status of your Paid Apps on app store connect? Make sure that it shall be green and active.
Here are some resources which can guide you as mentioned in the documentation itself.
* https://react-native-iap.dooboolab.com/docs/intro
* https://medium.com/dooboolab/announcing-react-native-iap-hooks-96c7ffd3f19a

Also I would suggest using hooks as it is more clean, clear and efficient way to implement RNIap. Just for a quick comparison all the heavy lifting is being handled by hook itself. Basically it handles all the async/await or callbacks or promises operation on its own.

**Comment by siddharthpvi** [comment]
App is on Testflight now

**Comment by siddharthpvi** [comment]
??

**Comment by cbothra** [comment]
> App is on Testflight now
It would not make any difference until the status of your paid app is *Active* (in green).

**Comment by siddharthpvi** [comment]
Do you mean have to live app?

**Comment by cbothra** [comment]
No! in the app store connect. 
1. Login to https://appstoreconnect.apple.com/
2. Select Agreements, Tax, and Banking
3. Once you are on agreements screen, there would be two options Free Apps & Paid Apps.
4. Fill out all the required agreement, taxes and banking details as shown in the attached screenshot.
![Screenshot 2022-01-03 at 2 38 47 PM](https://user-images.githubusercontent.com/1784356/147914773-ad8818c6-71c1-41a2-88f1-060039c05be6.png)

**Comment by siddharthpvi** [comment]
Thanks will check it

**Comment by andresesfm** [comment]
Thank you for the info @cbothra

**Comment by ColtraneNadler** [comment]
Was this issue ever resolved? I don't see @cbothra saying it was resolved, just explaining how to submit pertinent documents. I am currently experiencing the same issue of an empty array in result of getSubscription method. 
Any steps to solve would be helpful. I have created the skus in my account.

**Comment by TahseenSust** [comment]
I faced the same issue, despite my app's paid status being active. After some troubleshooting, I decided to check the country availability for the subscription services. Upon reviewing the supported regions, I realized that my location was not included in the list of eligible countries for subscription access. To resolve this, **I turned on a VPN and set it to a country where the subscription is available. After doing this and rerunning the app, the expected subscription array was successfully retrieved**. However, I wonder if there's any other solution or workaround that doesn't require using a VPN to access the subscription. Any suggestions?
